main/document/document.php

```diff
@@ -1110,68 +1137,96 @@ function convertModal (id, format) {
-                            'confirmation'
+                            get_lang('UrlAlreadyExists'),
+                            'error'
                         )
                     );
                 }
-
                 // Set the current path
                 $curdirpath = $_POST['move_to'];
                 $curdirpathurl = urlencode($_POST['move_to']);
             } else {
-                if ($fileExist) {
+                $real_path_target = $base_work_dir.$moveTo.'/'.basename($document_to_move['path']);
+                $fileExist = false;
+                if (file_exists($real_path_target)) {
+                    $fileExist = true;
+                }
+                if (move($base_work_dir.$document_to_move['path'], $base_work_dir.$moveTo)) {
+                    DocumentManager::updateDbInfo(
+                        'update',
+                        $document_to_move['path'],
+                        $moveTo.'/'.basename($document_to_move['path'])
+                    );
+
+                    //update database item property
+                    $doc_id = $_POST['move_file'];
                     if (is_dir($real_path_target)) {
-                        $message = Display::return_message(get_lang('DirExists'), 'error');
+                        api_item_property_update(
+                            $courseInfo,
+                            TOOL_DOCUMENT,
+                            $doc_id,
+                            'FolderMoved',
+                            api_get_user_id(),
+                            $group_properties,
+                            null,
+                            null,
+                            null,
+                            $sessionId
+                        );
+                        Display::addFlash(Display::return_message(get_lang('DirMv'), 'confirmation'));
                     } elseif (is_file($real_path_target)) {
-                        $message = Display::return_message(get_lang('FileExists'), 'v');
+                        api_item_property_update(
+                            $courseInfo,
+                            TOOL_DOCUMENT,
+                            $doc_id,
+                            'DocumentMoved',
+                            api_get_user_id(),
+                            $group_properties,
+                            null,
+                            null,
+                            null,
+                            $sessionId
+                        );
+                        Display::addFlash(
+                            Display::return_message(
+                                get_lang('DocMv'),
+                                'confirmation'
+                            )
+                        );
                     }
-                    Display::addFlash($message);
+
+                    // Set the current path
+                    $curdirpath = $_POST['move_to'];
+                    $curdirpathurl = urlencode($_POST['move_to']);
```

Comment:
This one is dangerous. I'm not sure the later use of $curdirpath or $curdirpathurl is safe in all cases. Please check that it doesn't contain dots for example. You can use the security.lib.php (Security class) methods for that.